editor/import/resource_importer_imagefont.cpp

```diff
@@ -267,23 +283,42 @@ Error ResourceImporterImageFont::import(ResourceUID::ID p_source_id, const Strin
 			if (end == -1) {
 				end = start;
 			}
+			if (!list.is_empty() && end != list[list.size() - 1]) {
+				list.push_back(end);
+			}
 
 			if (start == -1) {
 				WARN_PRINT(vformat("Invalid range: \"%s\"", range));
 				continue;
 			}
 		}
 
-		for (int32_t idx = MIN(start, end); idx <= MAX(start, end); idx++) {
-			ERR_FAIL_COND_V_MSG(pos >= count, ERR_CANT_CREATE, "Too many characters in range, should be " + itos(columns * rows));
-			int x = pos % columns;
-			int y = pos / columns;
-			font->set_glyph_advance(0, chr_height, idx, Vector2(chr_width + chr_adv, 0));
-			font->set_glyph_offset(0, Vector2i(chr_height, 0), idx, Vector2i(0, -0.5 * chr_height) + chr_off);
-			font->set_glyph_size(0, Vector2i(chr_height, 0), idx, Vector2(chr_width, chr_height));
-			font->set_glyph_uv_rect(0, Vector2i(chr_height, 0), idx, Rect2(img_margin.position.x + chr_cell_width * x + char_margin.position.x, img_margin.position.y + chr_cell_height * y + char_margin.position.y, chr_width, chr_height));
-			font->set_glyph_texture_idx(0, Vector2i(chr_height, 0), idx, 0);
-			pos++;
+		if (!list.is_empty()) {
+			ERR_FAIL_COND_V_MSG(list.size() > remaining, ERR_CANT_CREATE, vformat("Too many characters in range, should be %d.", columns * rows));
```

Comment:
```suggestion
			ERR_FAIL_COND_V_MSG(list.size() > remaining, ERR_CANT_CREATE, vformat("Too many characters in range, got %d but expected be %d.", list.size(), columns * rows));
```

Reply:
I'll change it to `vformat("Too many characters in range \"%s\", got %d but expected be %d.", range, size, remaining)` to show which of the ranges is the issue, otherwise it will be misleading and show size of the specific range with total character count.